### PR TITLE
SWP-676 - Implemented article preview for users with special privileges

### PR DESCRIPTION
**Pull request by takeit**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             | yes
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-676
| License                   | AGPLv3

Article preview is based on user permissions. If a user has, for example, `ROLE_EDITOR` role, he/she can preview article using url: `domain.com/<section>/<article-slug>`.

For example, if you created an article that has a slug `test-article` and is assigned to route `news`, it will be available for preview under url: `/news/test-article` only when user has `ROLE_CAN_VIEW_NON_PUBLISHED` role. In other cases 404 error will be thrown.

Preview url of an article can be taken from articles API (`/content/articles/`), from `_links` - `online` link and loaded in an iframe for preview.

TODO:
- [x] functional tests
- [x] documentation

### CODE REVIEW

File: docs/cookbooks/developers/article_preview.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
++-----------------------------------------------+---------------------------------+
+| ``ROLE_EDITOR``                               |               Yes               |
++-----------------------------------------------+---------------------------------+
+| ``ROLE_INTERNAL_API``                         |               Yes               |
++-----------------------------------------------+---------------------------------+
+| ``ROLE_ADMIN``                                |               Yes               |
++-----------------------------------------------+---------------------------------+
+| ``ROLE_SUPER_ADMIN``                          |               Yes               |
++-----------------------------------------------+---------------------------------+
+| ``IS_AUTHENTICATED_ANONYMOUSLY``              |               No                |
```
Comment (ahilles107) [inline docs/cookbooks/developers/article_preview.rst]: i would not list those not `eligible` as it requires more maintenance work.

Comment (takeit) [inline docs/cookbooks/developers/article_preview.rst]: I added it as an example only to list that default anonymous user wont have access to it

Comment (takeit) [inline docs/cookbooks/developers/article_preview.rst]: I can replace it with just `Anonymous` or write below that Anon users won't have access to the preview by default.

Comment (ahilles107) [inline docs/cookbooks/developers/article_preview.rst]: i would just list those roles who can preview unpublished content.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,7 +7,8 @@ in 0.1 minor versions.
 
 To get the diff for a specific change, go to https://github.com/superdesk/web-publisher/commit/XXX where XXX is the change hash
 
-* 0.5.0
+* 0.5.0-3
```
Comment (ahilles107) [inline CHANGELOG.md]: we should start new part for 0.6.0 with new features (as it's targeted next release).

**Pull request merged**